components/dashboard/src/workspaces/Workspaces.tsx

```diff
@@ -47,14 +47,18 @@ const WorkspacesPage: FunctionComponent = () => {
     const { filteredActiveWorkspaces, filteredInactiveWorkspaces } = useMemo(() => {
         const filteredActiveWorkspaces = activeWorkspaces.filter(
             (info) =>
-                `${info.name}${info.id}${info.contextUrl}${info.status?.gitStatus?.cloneUrl}${info.status?.gitStatus?.branch}`
+                `${info.metadata!.name}${info.id}${info.metadata!.originalContextUrl}${
```

Comment:
Is it always guaranteed that `metadata` is present? 
Reading `metadata!` a lot let me think if we can add an assertion when setting the `info` value.

Reply:
@AlexTugarev  Yes, we agree on it in this PR https://github.com/gitpod-io/gitpod/pull/19022#discussion_r1390887472

cc @akosyakov